Lib/_pyrepl/console.py

```diff
@@ -175,6 +187,7 @@ def _excepthook(self, typ, value, tb):
     def runsource(self, source, filename="<input>", symbol="single"):
         try:
             tree = ast.parse(source)
+            self.check_barry_as_FLUFL(tree)
```

Comment:
This should be after `except`, I guess.

Reply:
I also think that we don't need to do an extra `walk` in this function. We already have `for stmt in tree.body:` several lines below. And `from __future__ import ...` can be only a top-level import. So, we only check `tree.body`. Not nested imports.

This way it would be faster.

Also, why does this import need special treatment? `from __future__ import annotations` works with no special handling?